### PR TITLE
Upgrade to latest Closure Compiler

### PR DESCRIPTION
**Pull request by vandenoever**
A few annotation errors were uncovered and fixed.

### CODE REVIEW

**Comment by peitschie** [comment]
Fancy seeing you around these parts still!

Do you want review feedback on this? 😸

**Comment by thz** [comment]
are you talking to jos or good ol ko-ci? ;-)

**Comment by peitschie** [comment]
> are you talking to jos or good ol ko-ci? ;-)

Both? :smile:

**Comment by vandenoever** [comment]
@peitschie yes please
Glad to see you too.

File: webodf/lib/runtime.js [hunk 1]
```diff
@@ -1834,7 +1834,8 @@ var webodf = {};
             }
             runtime.setCurrentDirectory(path);
             function inner_run() {
-                var script, path, args, argv, result; // hide variables
+                // var script, path, args, argv, result; // hide variables
```
Comment (peitschie) [inline webodf/lib/runtime.js]: Could combine up to line 1841 into this. No real advantage to keeping the declare/assign separate

File: webodf/lib/ops/OpApplyHyperlink.js [hunk 2]
```diff
@@ -101,7 +101,8 @@ ops.OpApplyHyperlink = function OpApplyHyperlink() {
             linkNode = createHyperlink(ownerDocument, hyperlink);
             node.parentNode.insertBefore(linkNode, node);
             linkNode.appendChild(node);
-            if (modifiedParagraphs.indexOf(paragraph) === -1) {
+            if (paragraph !== null
```
Comment (peitschie) [inline webodf/lib/ops/OpApplyHyperlink.js]: I'm a smidge torn here, as a null paragraph would be very unexpected at this point. Would maybe be better to just hard-case the return type of the ``odfUtils.getParagraphElement`` call above to ``!Element``?

File: webodf/tests/odf/LayoutTests.js [hunk 3]
```diff
@@ -211,6 +211,9 @@ odf.LayoutTests = function LayoutTests(runner) {
             // This enables cross-browser tests to be written for cases where Chrome & FF perform different conversions
             // on the computed value (e.g., vertical-align with a %, FF converts to pixels, while Chrome returns the
             // original string).
+            if (typeof a === "number") {
```
Comment (peitschie) [inline webodf/tests/odf/LayoutTests.js]: Would it be safer to invert this? E.g., ``if (typeof a !== "string")``?

**Comment by peitschie** [comment]
I'm impressed at how few lines of code were needed to upgrade this. Looks great @vandenoever !

Given how minor things are, happy for you to :shipit: if you don't feel they're worth tweaking.

File: webodf/lib/ops/OpApplyHyperlink.js (hunk 2 above)
Comment (vandenoever) [inline webodf/lib/ops/OpApplyHyperlink.js]: getParagraphElement can take a ?Node. If that's null, it's passed straight through. So I prefer to keep the signature of getParagraphElement as it is.

File: webodf/lib/runtime.js (hunk 1 above)
Comment (vandenoever) [inline webodf/lib/runtime.js]: Fixed.

File: webodf/tests/odf/LayoutTests.js (hunk 3 above)
Comment (vandenoever) [inline webodf/tests/odf/LayoutTests.js]: The type of a is !string|!number but it cannot hurt to be more careful. Fixed.

**Comment by vandenoever** [comment]
@peitschie Thanks for the kind review.
I've used two suggestions.

**Pull request merged**